### PR TITLE
[PLANG-638] Datadog APM support

### PR DESCRIPTION
**Pull request by imrekel**

### CODE REVIEW

File: actions/app.go [hunk 1]
```diff
@@ -90,6 +92,7 @@ func App() *buffalo.App {
 			Env:         configs.GetENV(),
 			SessionName: "_addons-firebase-testlab_session",
 		})
+		app.Use(tracing.OpenTracing(tracer))
```
Comment (gaborszakacs) [inline actions/app.go:95]: Why do we define this here? Other middlewares seem to be injected around line 188.

File: tracing/opentracing.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
+// this is a customized version of https://github.com/bketelsen/buffet
```
Comment (gaborszakacs) [inline tracing/opentracing.go:1]: What was changed? Maybe it's worth mentioning that in the comment too.

File: tracing/opentracing.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	olog "github.com/opentracing/opentracing-go/log"
+)
+
+var tracer opentracing.Tracer
+
+// OpenTracing is a buffalo middleware that adds the necessary
+// components to the request to make it traced through OpenTracing.
+// Initialize it by passing in an opentracing.Tracer.
+func OpenTracing(tr opentracing.Tracer) buffalo.MiddlewareFunc {
+	tracer = tr
```
Comment (gaborszakacs) [inline tracing/opentracing.go:22]: Setting a global variable by creating a middleware is a bit unexpected side effect, I'd rather have an explicit function for that, but if it was like that in the original, I can live with it.

File: actions/app.go (hunk 1 above)
Comment (imrekel) [reply actions/app.go:95]: Moving it there - I just put it to the earliest possible place but you are right

File: tracing/opentracing.go (hunk 3 above)
Comment (imrekel) [reply tracing/opentracing.go:22]: yep, this is how it was in the original package

**Review by gaborszakacs — APPROVED** [review]

**Pull request merged**